docs/tools/debug/debugging_mbed_os_apps.md

```diff
@@ -1,4 +1,4 @@
-# Debugging Arm Mbed OS applications
+# Debugging methods for Arm Mbed OS applications
```

Comment:
Comment for line 5: this sentence should be removed or a link to debug_with_printf.md should be provided

Reply:
Added links to printf and serial communication.